R/setup.R

```diff
@@ -394,40 +369,29 @@ AgProductionTechnology_setup <- function(aLandLeaf, aAgData, aScenarioInfo) {
       agProdChange
   }
 
-  cost %>%
-    filter(AgProductionTechnology == name) ->
-    cost
-
   # Loop through all periods and read in data
   for(y in YEARS[[aScenarioInfo$mScenarioType]]) {
     per <- get_yr_to_per(y, aScenarioInfo$mScenarioType)
 
     # Only read in mCalOutput data for calibration periods
     if(per <= TIME.PARAMS[[aScenarioInfo$mScenarioType]]$FINAL_CALIBRATION_PERIOD) {
-      # Get calOutput for this period combination
-      calOutput %>%
-        filter(year == y) ->
-        currCalOutput
 
       # Set calOutput
-      if(nrow(currCalOutput)) {
-        aLandLeaf$mCalOutput[per] <- as.numeric(currCalOutput[[c("calOutputValue")]])
+      if(y %in% unique(calOutput$year) & name %in% unique(calOutput$AgProductionTechnology)) {
```

Comment:
Same comment as above.  `unique` can be a bit expensive.  Probably the cost is trivial compared to the other stuff we're doing, but I think the statement is also a bit easier to read without the extra statement, so it's a win either way.
```suggestion
      if(y %in% calOutput$year & name %in% calOutput$AgProductionTechnology) {
```